subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Utility functions to be used by the Wi-Fi subsytem.
+ */
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_wifi_utils, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/wifi.h>
+#include <zephyr/net/wifi_utils.h>
+
+
+static enum wifi_frequency_bands wifi_utils_map_band_str_to_idx(char *band_str)
+{
+	enum wifi_frequency_bands band = WIFI_FREQ_BAND_UNKNOWN;
+
+	if (!strcmp(band_str, "2")) {
+		band = WIFI_FREQ_BAND_2_4_GHZ;
+	} else if (!strcmp(band_str, "5")) {
+		band = WIFI_FREQ_BAND_5_GHZ;
+	} else if (!strcmp(band_str, "6")) {
+		band = WIFI_FREQ_BAND_6_GHZ;
+	}
+
+	return band;
+}
+
+
+static bool wifi_utils_validate_chan_2g(uint16_t chan)
+{
+	if ((chan >= 1) && (chan <= 14)) {
+		return true;
+	}
+
+	return false;
+}
+
+
+static bool wifi_utils_validate_chan_5g(uint16_t chan)
+{
+	uint16_t valid_5g_chans[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100, 104, 108,
+		112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 159, 161, 163, 165,
+		167, 169, 171, 173, 175, 177};
+	uint16_t i = 0;
+
+	for (i = 0; i < ARRAY_SIZE(valid_5g_chans); i++) {
+		if (chan == valid_5g_chans[i]) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+
+static bool wifi_utils_validate_chan_6g(uint16_t chan)
+{
+	if (((chan >= 1) && (chan <= 233) && (!((chan - 1)%4))) ||
+	    (chan == 2)) {
+		return true;
+	}
+
+	return false;
+}
+
+
+static bool wifi_utils_validate_chan(uint8_t band,
+				     uint16_t chan)
+{
+	bool result  = false;
+
+	switch (band) {
+	case WIFI_FREQ_BAND_2_4_GHZ:
+		result = wifi_utils_validate_chan_2g(chan);
+		break;
+	case WIFI_FREQ_BAND_5_GHZ:
+		result = wifi_utils_validate_chan_5g(chan);
+		break;
+	case WIFI_FREQ_BAND_6_GHZ:
+		result = wifi_utils_validate_chan_6g(chan);
+		break;
+	default:
+		NET_ERR("Unknown band: %d", band);
+		break;
+	}
+
+	return result;
+}
+
+
+static int wifi_utils_get_all_chans_in_range(uint16_t chan_start,
+		uint16_t chan_end,
+		uint16_t chan[__WIFI_FREQ_BAND_AFTER_LAST][WIFI_CHANNEL_MAX],
+		uint8_t band_idx,
+		uint8_t *chan_idx)
+{
+	uint16_t i = 0;
+	bool start = false;
+	bool end = false;
+	uint16_t valid_5g_chans[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100, 104, 108,
```

Comment:
If we are not changing this, then we could mark it const

Reply:
Made this a const.